cookbooks/bcpc-hadoop/recipes/disks.rb

```diff
@@ -31,17 +36,22 @@
       action [:enable, :mount]
     end
   end
-  if node.roles.include? "BCPC-Hadoop-Head" and node[:bcpc][:hadoop][:head_disks] != nil then
-    mnt_idxs = node[:bcpc][:hadoop][:head_disks].map do |dsk| 
-      node[:bcpc][:hadoop][:disks].index(dsk)
+
+  # is our role included in the list
+  if not (node[:bcpc][:hadoop][:disks][:disk_exclude_roles] & node.roles).empty? then 
```

Comment:
Shouldn't you use `&&` for logical `AND` as mentioned [here](http://www.tutorialspoint.com/ruby/ruby_operators.htm).